### ISSUE TITLE
Cannot retrieve built in policy definitions by name.

### ISSUE THREAD

**Issue by Nepomuceno** [terraform-providers/terraform-provider-azurerm#9073]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

terraform version 0.13.5

### Affected Resource(s)

* `azurerm_policy_definition`

### Terraform Configuration Files

```hcl
provider "azurerm" {
  version = "=2.33.0"
  features {}
}

data "azurerm_policy_definition" "mfa_enabled" {
  name = "aa633080-8b72-40c4-a2d7-d00c03e80bed"
}
```

### Expected Behavior

Policy was found and returned.

### Actual Behavior

```bash
Error: reading Policy Definition "aa633080-8b72-40c4-a2d7-d00c03e80bed": policy.DefinitionsClient#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="PolicyDefinitionNotFound" Message="The policy definition 'aa633080-8b72-40c4-a2d7-d00c03e80bed' could not be found."

  on main.tf line 6, in data "azurerm_policy_definition" "mfa_enabled":
   6: data "azurerm_policy_definition" "mfa_enabled" {
```

### Steps to Reproduce

1. `terraform apply`

**Comment by neil-yechenwei** [comment]
Thanks for opening this issue. I assume you are requesting a [new data source](https://docs.microsoft.com/en-us/rest/api/resources/policydefinitions/getbuiltin) for retrieving builtin policy definition since [azurerm_policy_definition data source](https://docs.microsoft.com/en-us/rest/api/resources/policydefinitions/get) only can retrieve a policy definition in a subscription.

**Comment by magodo** [comment]
I assume this is already fixed in #9078. So close it for now.
